src/Maths/math.vector.ts

```diff
@@ -5134,6 +5156,97 @@ export class Matrix {
+        // First ensure that two vectors are valid and approximately orthogonal.
+        if (forward.length() < Epsilon || up.length() < Epsilon || Math.abs(Vector3.Dot(forward, up)) >= Epsilon) {
+            result.copyFrom(Matrix.IdentityReadOnly);
+        }
+
+        // Next normalize the two inputted vectors and take the cross product to find the third orthonormal vector that defines the rotation.
+        const forwardNorm = MathTmp.Vector3[0];
+        const upNorm = MathTmp.Vector3[1];
+        const left = MathTmp.Vector3[2];
+        forward.normalizeToRef(forwardNorm);
+        forwardNorm.scaleInPlace(-1);
+        up.normalizeToRef(upNorm);
+        Vector3.CrossToRef(upNorm, forwardNorm, left);
+        left.normalize();
+
+        // Generate the rotation matrix.
+        Matrix.FromValuesToRef(
+            left._x, left._y, left._z, 0.0,
+            upNorm._x, upNorm._y, upNorm._z, 0.0,
+            forwardNorm._x, forwardNorm._y, forwardNorm._z, 0.0,
+            0, 0, 0, 1.0,
+            result
+        );
+    }
+
+     /**
+     * Gets a new rotation matrix used to rotate an entity so as it looks in the direction specified by forward from the eye position, the up Vector3 being oriented like "up".
+     * This function works in right handed mode
+     * @param forward defines the forward direction
+     * @param up defines the up vector for the entity
+     * @returns the new matrix
+     */
+    public static LookDirectionRH(forward: DeepImmutable<Vector3>, up: DeepImmutable<Vector3>): Matrix {
+        var result = new Matrix();
+        Matrix.LookDirectionRHToRef(forward, up, result);
+        return result;
+    }
+
+    /**
+     * Sets the given "result" Matrix to a rotation matrix used to rotate an entity so that it looks in the direction of forward, the up vector3 being oriented like "up".
+     * This function works in right handed mode
+     * @param forward defines the forward direction
+     * @param up defines the up vector for the entity
+     * @param result defines the target matrix
+     */
+    public static LookDirectionRHToRef(forward: DeepImmutable<Vector3>, up: DeepImmutable<Vector3>, result: Matrix): void {
+        // First ensure that two vectors are valid and approximately orthogonal.
+        if (forward.length() < Epsilon || up.length() < Epsilon || Math.abs(Vector3.Dot(forward, up)) >= Epsilon) {
+            result.copyFrom(Matrix.IdentityReadOnly);
+        }
+
+        // Next normalize the two inputted vectors and take the cross product to find the third orthonormal vector that defines the rotation.
+        const forwardNorm = MathTmp.Vector3[0];
+        const upNorm = MathTmp.Vector3[1];
+        const right = MathTmp.Vector3[2];
+        forward.normalizeToRef(forwardNorm);
+        up.normalizeToRef(upNorm);
+        Vector3.CrossToRef(upNorm, forwardNorm, right);
+        right.normalize();

```

Comment:
You can save this normalization: as up and forward are orthogonal and of unit length, so their cross product is also of unit length.

Reply:
Good catch, done.